src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -1165,14 +1165,23 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
 
     private selectNewKernel() {
         // This is handled by a command.
-        this.commandManager.executeCommand(Commands.SwitchJupyterKernel, {
-            identity: this.notebookIdentity.resource,
-            resource: this.owningResource,
-            currentKernelDisplayName:
-                this.notebookMetadata?.kernelspec?.display_name ||
-                this.notebookMetadata?.kernelspec?.name ||
-                getDisplayNameOrNameOfKernelConnection(this._notebook?.getKernelConnection())
-        });
+        this.commandManager
+            .executeCommand(Commands.SwitchJupyterKernel, {
+                identity: this.notebookIdentity.resource,
+                resource: this.owningResource,
+                currentKernelDisplayName:
+                    this.notebookMetadata?.kernelspec?.display_name ||
+                    this.notebookMetadata?.kernelspec?.name ||
+                    getDisplayNameOrNameOfKernelConnection(this._notebook?.getKernelConnection())
+            })
+            .then(async () => {
+                // Reset our file in the kernel.
+                const fileInKernel = this.fileInKernel;
```

Comment:
I think it would actually be better to do this in the 'kernelChangedHandler' function. That would handle the case where this function isn't called (like if the command is executed from the palette).

Reply:
Not a fan of this, we have a lot of other code where we initialize the kernel once started/restarted (setting cwd, matplot lib options, jedi options, & more )
I'd prefer this was done in the same place where we do the other stuff.